juju/juju
# Remove Mongo-based leases logic

Here we finish removing all dealings with Mongo as regards leases. Changes included are:
- Removal of some dangling trapdoor declarations that are no longer used following #14653.
- Removal of all state logic for leases, including the Mongo collection.
- Removal of the lease notify functionality from workers, which was previously used to update the Mongo collection upon lease claim/expiry.
- Removal of state dependencies from lease-related workers that no longer need it.
- Population of application leaders in exported models from the Raft-backed lease worker instead of Mongo.

## QA steps

- Deploy a multi-unit workload and observe which unit becomes leader.
- Run an action on the leader by targeting `<application>/leader`.
- Remove the leader unit, observe that a new unit becomes leader, and run the leader action again.
- Migrate the model to another controller and check that it completes successfully.

## Documentation changes

None.

## Bug reference

N/A
